python_ta/z3/z3_parser.py

```diff
@@ -296,3 +299,22 @@ def parse_arguments(self, node: astroid.Arguments) -> dict[str, z3.ExprRef]:
                 z3_vars[arg.name] = self.parse(arg)
 
         return z3_vars
+
+    def parse_call(self, node: nodes.Call) -> Union[z3.ExprRef, list[z3.ExprRef]]:
+        if not isinstance(node.func, nodes.Name):
```

Comment:
Overall I prefer using a single if statement with multiple branches. (I can tell you don't, but this is an example of sticking to the preferred style of the existing codebase.) Please modify this function to use a single if statement with multiple branches for each of the cases you identified.